File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.cpp
#include <string>
#include <vector>
#include <map>
#include <iostream>
#include <algorithm>

using namespace std;

int solution(int k, vector<int> tangerine) {
    int answer = 0;
    
    map<int, int> m;
    
    for (int i = 0; i < tangerine.size(); i++) {
        m[tangerine[i]]++;
    }
    
    vector<int> v;
    
    for (auto iter : m) {
        v.push_back(iter.second);
    }
    
    sort(v.begin(), v.end(), greater<int>());
    
    for (int i = 0; i < v.size(); i++) {
        k -= v[i];
        answer++;
        if (k <= 0) {
            break;
        }
    }
    
    return answer;
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
from collections import Counter

def solution(k, tangerine):
    answer = 0
    
    # 귤 크기별 개수 세기
    counter = Counter(tangerine)
    
    # 개수 기준으로 내림차순 정렬
    sorted_counts = sorted(counter.values(), reverse=True)
    
    # 많이 나온 귤부터 담기
    total = 0
    for cnt in sorted_counts:
        total += cnt
        answer += 1
        if total >= k:
            break
    
    return answer
